shared/test/test_assets.rb

```diff
@@ -0,0 +1,98 @@
+require 'minitest/autorun'
+require 'rack/test'
+require File.expand_path '../../../deployment', __FILE__
+require File.expand_path '../../middleware/assets_api', __FILE__
+require File.expand_path '../../middleware/channels_api', __FILE__
+
+ENV['RACK_ENV'] = 'test'
+
+class AssetsTest < Minitest::Unit::TestCase
+
+  def create_channel
+    @channels.post '/v3/channels', {}.to_json, 'CONTENT_TYPE' => 'application/json;charset=utf-8'
+    @channel_id = @channels.last_response.location.split('/').last
+  end
+
+  def delete_channel
+    @channels.delete "/v3/channels/#{@channel_id}"
+    #assert @channels.last_response.successful?
+  end
+
+  def list()
+    @assets.get("/v3/assets/#{@channel_id}").body
+  end
+
+  def put(filename, body, content_type)
+    @assets.put("/v3/assets/#{@channel_id}/#{filename}", body, 'CONTENT_TYPE' => content_type).body
+  end
+
+  def get(filename)
+    @assets.get "/v3/assets/#{@channel_id}/#{filename}"
+    @assets.last_response.body
+  end
+
+  def delete(filename)
+    @assets.delete "/v3/assets/#{@channel_id}/#{filename}"
+  end
+
+  def assert_fileinfo_equal(expected, actual)
+    assert_equal(Hash, actual.class)
+    assert_equal(expected['filename'], actual['filename'])
+    assert_equal(expected['category'], actual['category'])
+    assert_equal(expected['size'], actual['size'])
+  end
+
+  def test_assets
+    # The Assets API does not *currently* need to share a cookie jar with the Channels API,
+    # but it may once we restrict put, delete and list operations to the channel owner.
```

Comment:
I spent some time on this when implementing the properties and tables api tests and eventually realized I didn't need to share cookie jars to make the tests work. It will require some further investigation but seems like it should be possible. If not there are a few other ways we could continue to work around it.